numba/mviewbuf.c

```diff
@@ -1,5 +1,199 @@
 #include "_pymodule.h"
 
+/**
+ * A BufferProxy wraps a read-only buffer in a writable buffer. This is required
+ * for NumPy void and record types, which only expose a read-only buffer. Note
+ * that wrapping a read-only buffer may have some surprising effects on buffers
+ * which expect the data from their read-only buffer not to be modified.
+ */
+typedef struct {
+    PyObject_HEAD
+    PyObject *wrapped;
+    int ob_exports;
+    Py_buffer wrapped_buf;
+#if (PY_MAJOR_VERSION < 3)
+    void *wrapped_ptr;
+    Py_ssize_t wrapped_len;
+    int ob_old_exports;
+#endif
+} BufferProxyObject;
+
+static int
+BufferProxyObject_getbuffer(BufferProxyObject *self, Py_buffer *view,
+                            int flags) {
+    int supported_flags = PyBUF_WRITABLE|PyBUF_ND|PyBUF_STRIDES|PyBUF_FORMAT;
+    int wrapped_flags = supported_flags & ~PyBUF_WRITABLE;
+
+    if (flags & ~supported_flags) {
+        PyErr_SetString(PyExc_BufferError, "Unsupported flags for BufferProxy");
+        view->obj = NULL;
+        return -1;
+    }
+
+    if (self->ob_exports == 0) {
+        if(-1 == PyObject_GetBuffer(self->wrapped, &(self->wrapped_buf),
+                                    wrapped_flags)) {
+            PyErr_SetString(PyExc_BufferError,
+                            "Could not get buffer for wrapped (new protocol)");
+            view->obj = NULL;
+            return -1;
+        }
+    }
+
+    if(-1 == PyBuffer_FillInfo(view, (PyObject*) self, self->wrapped_buf.buf,
+                               self->wrapped_buf.len, 0, supported_flags)) {
+        PyBuffer_Release(&(self->wrapped_buf));
+        view->obj = NULL;
+        return -1;
+    }
+
+    self->ob_exports += 1;
+    return 0;
+}
+
+static void
+BufferProxyObject_releasebuffer(BufferProxyObject *self, Py_buffer *view) {
```

Comment:
For the record, have you checked this function gets called? It seems that by always forcing the child view's "obj" field to NULL above, there is no way releasing the child buffer can call into this.